rust/src/domains/ffi.rs

```diff
@@ -358,9 +358,9 @@ pub extern "C" fn opendp_domains__map_domain(
     match (&key_domain.type_.contents, &value_domain.type_.contents) {
         (TypeContents::GENERIC { name: "AtomDomain", .. }, TypeContents::GENERIC { name: "AtomDomain", .. }) => 
             dispatch!(monomorphize, [(key_domain.carrier_type, @hashable), (value_domain.carrier_type, @primitives)], (key_domain, value_domain)),
-        (TypeContents::GENERIC { name: "AtomDomain", .. }, TypeContents::PLAIN("UserDomain")) => 
+        (TypeContents::GENERIC { name: "AtomDomain", .. }, TypeContents::PLAIN("ExtrinsicDomain")) => 
             dispatch!(monomorphize_extrinsic, [(key_domain.carrier_type, @hashable)], (key_domain, value_domain)),
-        _ => fallible!(FFI, "MapDomain constructor only supports AtomDomain or UserDomain inner domains")
+        _ => fallible!(FFI, "Value domain of MapDomain must be AtomDomain or ExtrinsicDomain (created via user_domain)"),
```

Comment:
I think `user_domain` is a python function, and isn't part of the rust code, so should it be in a rust error message? More often we worry about rust details bleeding through into python, but it happens the other way, too. This probably isn't something to get hung up about, but with possibly more language bindings we'd want to avoid error messages that hard code one particular naming convention. (In C# the function would probably be `UserDomain`.)